libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/OAuth2.java

```diff
@@ -638,7 +641,7 @@ private void populateFromJSON(JSONObject parsedResponse) {
                 firstName = parsedResponse.getString(FIRST_NAME);
                 lastName = parsedResponse.getString(LAST_NAME);
                 displayName = parsedResponse.getString(DISPLAY_NAME);
-                final JSONObject photos = parsedResponse.getJSONObject(PHOTOS);
+                final JSONObject photos = parsedResponse.optJSONObject(PHOTOS);
```

Comment:
Old code - if it can get null, we need to to optJSONObject not getJSONObject (which throws).
Because of the catch below, the code will continue with all the fields below the throwing line not set :-(